tools/sys_analyzer/sys_analyzer_linux.py

```diff
@@ -0,0 +1,704 @@
+            print( "usage: python3 sys_analyzer_linux.py [-v]")
+            print("-v = verbose")
+            sys.exit(0)
+
+        if (sys.argv[1]=="-v"):
+            GLOBAL_LOGLEVEL=loglevelcode.INFO
+
+    #HW media ready: processor,gpu ID (yes,no,advice)
+    print("--------------------------")
+    print("Hardware readiness checks:")
+    print("--------------------------")
+    sts=does_processor_have_gen_graphics()
+    if (sts<loglevelcode.SUCCESS): sys.exit(1)
+
+    #OS media ready: OS,glibc version,gcc version,nomodeset,gpuID (yes, no, advice)
+    print("--------------------------")
+    print("OS readiness checks:")
+    print("--------------------------")
+    sts=is_OS_media_ready()
+    if (sts<loglevelcode.SUCCESS): sys.exit(1)
+
+    #Media stack install correctness: vainfo, /dev/dri, check MSDK dirs, check OCL dirs, check MSDK/OCL funcs
+    print("--------------------------")
+    print("User and kernel mode drivers health checks:")
+    print("--------------------------")
+    #in video group
+    out=run_cmd("groups")
+    if ("video" in out):
+        print_info("user in 'video' group",loglevelcode.SUCCESS)
+    else:
+        print_info("user not in the 'video' group. Add with 'usermod -a -G video {user}'",loglevelcode.ERROR)
+
+    if pkgconfig.exists('libva'):
+        print_info("libva pkg-config found",loglevelcode.SUCCESS)
+    else:
+        print_info("libva package not found in default PKG_CONFIG locations. Trying again with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/",loglevelcode.INFO)
+        os.environ["PKG_CONFIG_PATH"] = "/opt/intel/mediasdk/lib64/pkgconfig/"
+        print_info("Trying again with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/",loglevelcode.INFO)
+        if pkgconfig.exists('libva'):
+            print_info("libva found with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/",loglevelcode.SUCCESS)
+        else:
+            print_info("libva not found with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/",loglevelcode.ERROR)
+
+    #vainfo: exists,correct iHD used
+    out=run_cmd("vainfo 2>&1")
+    if ("iHD_drv_video.so" in out):
+        print_info("Intel iHD used by libva",loglevelcode.INFO)
+    else:
+        print_info("libva not loading Intel iHD",loglevelcode.ERROR)
+    
+    if ("VAEntrypoint" in out):
+        print_info("vainfo reports valid codec entry points",loglevelcode.SUCCESS)
+    else:
+        print_info("vainfo not reporting codec entry points",loglevelcode.ERROR)       
+
+    #check i915 use
+    out=run_cmd("lspci -v -s 0:02.0")
+    if ("Kernel driver in use: i915" in out):
+        print_info("i915 driver in use by Intel video adapter",loglevelcode.INFO)
+    else:
+        print_info("Intel video adapter not using i915",loglevelcode.ERROR)
+
+    check_libdrm()
+
+    print("--------------------------")
+    print("Component Smoke Tests:")
+    print("--------------------------")
+    check_msdk_api()
```

Comment:
I wish to see successful output of the script....

Can you add a check whether HuC is available on the system w/ a note which features might be missing if it is not available?

Reply:
> I wish to see successful output of the script....

From SKL:
```
./sys_analyzer_linux.py
--------------------------
Hardware readiness checks:
--------------------------
 [ OK ] Processor name: Intel(R) Core(TM) i5-6260U CPU @ 1.80GHz
--------------------------
OS readiness checks:
--------------------------
 [ OK ] GPU visible to OS
--------------------------
User and kernel mode drivers health checks:
--------------------------
 [ ERROR ] user not in the 'video' group. Add with 'usermod -a -G video {user}'
 [ OK ] libva pkg-config found
 [ OK ] vainfo reports valid codec entry points
 [ OK ] /dev/dri/renderD128 connects to Intel i915
--------------------------
Component Smoke Tests:
--------------------------
 [ OK ] Media SDK HW API level:1.30

```

```
./sys_analyzer_linux.py -v
--------------------------
Hardware readiness checks:
--------------------------
 [ OK ] Processor name: Intel(R) Core(TM) i5-6260U CPU @ 1.80GHz
 [ INFO ] Intel Processor
 [ INFO ] Processor brand: Core
--------------------------
OS readiness checks:
--------------------------
 [ INFO ] GPU PCI id     : 1926
 [ INFO ] GPU description: SKL ULT GT3
 [ OK ] GPU visible to OS
 [ INFO ] no nomodeset in GRUB cmdline (good)
 [ INFO ] Linux distro   : CentOS Linux 7.5.1804
 [ INFO ] Linux kernel   : 5.0.3-1.el7.elrepo.x86_64
 [ INFO ] glibc version  : 2.17
 [ INFO ] gcc version    : 4.8.5-28) (>=4.8.2 suggested)
--------------------------
User and kernel mode drivers health checks:
--------------------------
 [ ERROR ] user not in the 'video' group. Add with 'usermod -a -G video {user}'
 [ OK ] libva pkg-config found
 [ INFO ] Intel iHD used by libva
 [ OK ] vainfo reports valid codec entry points
 [ INFO ] i915 driver in use by Intel video adapter
 [ OK ] /dev/dri/renderD128 connects to Intel i915
--------------------------
Component Smoke Tests:
--------------------------
 [ OK ] Media SDK HW API level:1.30
 [ INFO ] no OpenCL include files. Are Intel components installed?

```
From ICL:
```
./sys_analyzer_linux.py -v
--------------------------
Hardware readiness checks:
--------------------------
 [ OK ] Processor name: Genuine Intel(R) CPU 0000 @ 1.00GHz
 [ INFO ] Intel Processor
 [ INFO ] Processor not Xeon or Core.
--------------------------
OS readiness checks:
--------------------------
 [ INFO ] GPU PCI id     : 8A52
 [ INFO ] GPU description: ICL GT2
 [ OK ] GPU visible to OS
 [ INFO ] no nomodeset in GRUB cmdline (good)
 [ INFO ] Linux distro   : Ubuntu 18.04
 [ INFO ] Linux kernel   : 5.3.0
 [ INFO ] glibc version  : 2.27
 [ INFO ] gcc version    : 7.3.0 (>=4.8.2 suggested)
--------------------------
User and kernel mode drivers health checks:
--------------------------
 [ OK ] user in 'video' group
 [ INFO ] libva package not found in default PKG_CONFIG locations. Trying again with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/
 [ INFO ] Trying again with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/
 [ OK ] libva found with PKG_CONFIG_PATH=/opt/intel/mediasdk/lib64/pkgconfig/
 [ INFO ] Intel iHD used by libva
 [ OK ] vainfo reports valid codec entry points
 [ INFO ] i915 driver in use by Intel video adapter
 [ OK ] /dev/dri/renderD128 connects to Intel i915
--------------------------
Component Smoke Tests:
--------------------------
 [ OK ] Media SDK HW API level:1.30
 [ INFO ] no OpenCL include files. Are Intel components installed?

```